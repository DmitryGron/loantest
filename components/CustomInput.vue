<template>
  <div class="flex flex-col">
    <label>
      {{ label }}
    </label>
    <input
        v-model="model"
        class="text-[#0c0c0c] border-[#6b6d6e] border-[1.5px] rounded"
        type="text"
        :name="name"
      />
  </div>
</template>

<script>
export default {
  name: 'CustomInput',
  props: {
    label: {
      type: String,
      required: true,
    },
    value: {
      type: [Number, String],
      required: true,
    },
  },
  computed: {
    name() {
      return this.label.toLowerCase()
    },
    model: {
      get() {
        return this.value
      },
      set(value) {
        this.$emit('input', value)
      },
    },
  },
}
</script>
