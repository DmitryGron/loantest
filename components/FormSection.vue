<template>
  <div class="bg-white rounded border-gray-300 border-[2px] mx-10 p-10 shadow-md">
    <h1 class="text-[#0c0c0c] text-2xl font-semibold">Loan Decision Engine</h1>
    <form class="flex flex-col gap-4 mt-4" @submit.prevent="submitForm">
      <slot></slot>
      <button
        class="text-white bg-[#3498db] hover:bg-[#2980b9] py-2 rounded shadow-md transition-all duration-200"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</template>

<script>
export default {
  name: 'FormSection',
  methods: {
    submitForm() {
      this.$emit('submit');
    },
  },
};
</script>
