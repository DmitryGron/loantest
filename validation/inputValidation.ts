export const validatePersonalCode = (personalCode: string): boolean =>
  !(personalCode?.length !== 11 || isNaN(Number(personalCode)));
export const validateLoanAmount = (amount: number): boolean =>
  !(amount! < 2000 || amount > 10000 || isNaN(amount));
export const validateLoanPeriod = (period: number): boolean =>
  !(period < 12 || period > 60 || isNaN(period));
