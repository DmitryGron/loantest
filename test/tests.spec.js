import { shallowMount } from '@vue/test-utils'
import CustomInput from '~/components/CustomInput.vue'
import FormSection from '~/components/FormSection.vue'
import DecisionPage from '~/pages/index.vue'
import { validatePersonalCode, validateLoanAmount, validateLoanPeriod } from '~/validation/inputValidation'

describe('DecisionPage', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(DecisionPage);
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders correctly', () => {
    const wrapper = shallowMount(DecisionPage);
    expect(wrapper.element).toMatchSnapshot();
  })

  test('renders FormSection', () => {
    const wrapper = shallowMount(DecisionPage);
    expect(wrapper.findComponent(FormSection).exists()).toBe(true)
  })
});

describe('CustomInput', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(CustomInput, {
      propsData: {
        label: 'Test Label', // Provide the label prop here
        value: 'Test Value', // Provide the value prop here
      },
    });
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders correctly', () => {
    const wrapper = shallowMount(CustomInput, {
      propsData: {
        label: 'Test Label', // Provide the label prop here
        value: 'Test Value', // Provide the value prop here
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  })

  test('renders label correctly', () => {
    const wrapper = shallowMount(CustomInput, {
      propsData: {
        label: 'Test Label', // Provide the label prop here
        value: 'Test Value', // Provide the value prop here
      },
    });
    expect(wrapper.find('label').text()).toBe('Test Label');
  })

  test('renders value correctly', () => {
    const wrapper = shallowMount(CustomInput, {
      propsData: {
        label: 'Test Label', // Provide the label prop here
        value: 'Test Value', // Provide the value prop here
      },
    });
    expect(wrapper.find('input').element.value).toBe('Test Value');
  })

  test('emits input event when input value changes', async () => {
    const wrapper = shallowMount(CustomInput, {
      propsData: {
        label: 'Test Label', // Provide the label prop here
        value: 'Test Value', // Provide the value prop here
      },
    });
    await wrapper.find('input').setValue('New Value');
    expect(wrapper.emitted('input')).toBeTruthy();
  })
});

describe('validatePersonalCode', () => {
  test('returns true when personal code is valid', () => {
    const personalCode = "12345612345"
    expect(validatePersonalCode(personalCode)).toBe(true)
  })

  test('returns false when personal code is invalid', () => {
    const personalCode = '123456-1234'
    expect(validatePersonalCode(personalCode)).toBe(false)
  })

  test('returns false when personal code is invalid', () => {
    const personalCode = ''
    expect(validatePersonalCode(personalCode)).toBe(false)
  })

  test('returns false when personal code is invalid', () => {
    const personalCode = null
    expect(validatePersonalCode(personalCode)).toBe(false)
  })

  test('returns false when personal code is invalid', () => {
    const personalCode = 'qqqqqqqqqqq'
    expect(validatePersonalCode(personalCode)).toBe(false)
  })
})

describe('validateLoanAmount', () => {
  test('returns true when loan amount is valid', () => {
    const loanAmount = 2000
    expect(validateLoanAmount(loanAmount)).toBe(true)
  })

  test('returns false when loan amount is invalid', () => {
    const loanAmount = 1000
    expect(validateLoanAmount(loanAmount)).toBe(false)
  })

  test('returns false when loan amount is invalid', () => {
    const loanAmount = "aweqweqwe"
    expect(validateLoanAmount(loanAmount)).toBe(false)
  })

  test('returns false when loan amount is invalid', () => {
    const loanAmount = 11000
    expect(validateLoanAmount(loanAmount)).toBe(false)
  })
})

describe('validateLoanPeriod', () => {
  test('returns true when loan period is valid', () => {
    const loanPeriod = 12
    expect(validateLoanPeriod(loanPeriod)).toBe(true)
  })

  test('returns true when loan period is valid', () => {
    const loanPeriod = 60
    expect(validateLoanPeriod(loanPeriod)).toBe(true)
  })

  test('returns false when loan period is invalid', () => {
    const loanPeriod = 0
    expect(validateLoanPeriod(loanPeriod)).toBe(false)
  })

  test('returns false when loan period is invalid', () => {
    const loanPeriod = 61
    expect(validateLoanPeriod(loanPeriod)).toBe(false)
  })
})




