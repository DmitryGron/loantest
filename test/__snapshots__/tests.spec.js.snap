// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput renders correctly 1`] = `
<div
  class="flex flex-col"
>
  <label>
    
    Test Label
  
  </label>
   
  <input
    class="text-[#0c0c0c] border-[#6b6d6e] border-[1.5px] rounded"
    name="test label"
    type="text"
  />
</div>
`;

exports[`DecisionPage renders correctly 1`] = `
<div
  class="flex flex-col justify-center items-center mt-10 gap-5 w-full"
>
  <infosection-stub />
   
  <formsection-stub>
    <custominput-stub
      label="Personal Code:"
      required=""
      validate="function () { [native code] }"
      value=""
    />
     
    <custominput-stub
      label="Loan Amount (â‚¬):"
      required=""
      validate="function () { [native code] }"
      value="0"
    />
     
    <custominput-stub
      label="Loan Period (months):"
      required=""
      validate="function () { [native code] }"
      value="0"
    />
  </formsection-stub>
   
  <!---->
</div>
`;
